### PR TITLE
Remove Minimum Number of Parenthesis for Balanced String in python

### LINKED ISSUE #2154: Remove minimum parantheses to make Valid Parentheses
## 🚀 Feature

Given a string of '(' , ')' and lowercase english characters. 

The task is to remove the minimum number of parentheses so that the resulting parentheses string is valid.
A parentheses string is valid if and only if:
1.It is the empty string, contains only lowercase characters, or
2.It can be written as AB (A concatenated with B), where A and B are valid strings, or
3.It can be written as (A), where A is a valid string.

Example:
1. Input: "he((l)l)o)"
   Output: "he((l)l)o"
2. Input: ")))((("
    Output: ""

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes.

## Motivation

Helps in enhancing problem-solving.

## Pitch

Helps in understanding the basics of stacks.

### PR DESCRIPTION
**Pull request by nandinib1999**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

This PR resolves #2154

### CODE REVIEW

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
Few more changes: 

- Make README updates
- Use Comments above Line of Code not beside it
- Add Test-Cases and complexity at the end of the Code

**Comment by nandinib1999** [comment]
Not sure why the "DeepSource: Java (beta)" test is failing. :(

File: Python/other/check_valid_parenthesis_string.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        if i in final_indx:
+            pass
+        else:
+            op_str = op_str + s
+    return op_str
+
+
+if __name__ == '__main__':
+    # Output: he((l)l)o
+    remove_parenthesis('he((l)l)o)')
```
Comment (HarshCasper) [inline Python/other/check_valid_parenthesis_string.py]: Use Dynamic Input

File: Python/other/check_valid_parenthesis_string.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    # Time Complexity: O(n)
+    # Testcases
+    # (1) Input: (()))() Output: (())()
+    # (2) Input: neoalgo Output: neoalgo
+    # (3) Input: ())(() Output: ()()
+    # (4) Input: )))((( Output: ''
+    # (5) Input: (())() Output: (())()
```
Comment (HarshCasper) [inline Python/other/check_valid_parenthesis_string.py:86]: Use Multi-Line Strings

File: Python/other/check_valid_parenthesis_string.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+they contain the indices of unpaired opening and closing brackets.
+An output string is generated after skipping the characters at indices
+left in "stack" and "indx".
+
+Argument: string (Input string with parenthesis)
+Return: string (Output string with balanced parenthesis)
+'''
+
+
+def remove_parenthesis(string):
```
Comment (HarshCasper) [inline Python/other/check_valid_parenthesis_string.py:34]: Add a function docstring

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/other/check_valid_parenthesis_string.py (hunk 2 above)
Comment (nandinib1999) [reply Python/other/check_valid_parenthesis_string.py:86]: I had used multi-line string comment but DeepSource: Python gave an error of "unassigned string". That's why I moved to single-line comments.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**